src/diffusers/models/resnet.py

```diff
@@ -373,59 +408,89 @@ def __init__(
         conv_shortcut=False,
         dropout=0.0,
         temb_channels=512,
-        groups=32,
+        groups: Optional[int]=32,
         groups_out=None,
         pre_norm=True,
         eps=1e-6,
         non_linearity="swish",
-        time_embedding_norm="default",
+        norm1_scale_shift: bool = False, # if true, will use ada groupnorm (i.e. time embedding scale shift) instead of groupnorm
```

Comment:
Can we maybe call this:
```suggestion
        norm_1_type: str = "group", # or "ada_group"
```